File: bg/WCScala/src/main/scala/com/kafka/kafkaCP/KafkaProducer.java
package com.kafka.kafkaCP;

import org.apache.kafka.clients.producer.ProducerRecord;

import java.util.Properties;

public class KafkaProducer {

    public static void main(String[] args) {

        Properties props = new Properties();

        props.put("enable.idempotence",true);
        props.put("acks", "all"); // 当 enable.idempotence 为 true，这里默认为 all
        props.put("bootstrap.servers", "hdp-01:9092,hdp-02:9092,hdp-03:9092");
        props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
        props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");

        props.put("retries", 3);
        //Reduce the no of requests less than 0
        props.put("linger.ms", 1);
        //The buffer.memory controls the total amount of memory available to the producer for buffering.
        props.put("buffer.memory", 33550);

        org.apache.kafka.clients.producer.KafkaProducer<String, String> producer = new org.apache.kafka.clients.producer.KafkaProducer<>(props);

        for (int i=0;i<10;i++){
            producer.send(new ProducerRecord<String, String>("",""));
        }


    }
}

File: bg/shoppingMarket/NewSparkStreaming/src/main/scala/com/flyex/newSpark/KafkaProducer.scala
package com.flyex.newSpark

import java.util.Properties

import com.typesafe.config.ConfigFactory
import org.apache.kafka.clients.producer.{KafkaProducer, Producer, ProducerRecord}

object KafkaProducer {

  def main(args: Array[String]): Unit = {

    val load = ConfigFactory.load()

    val props = new Properties()
    props.put("bootstrap.servers",load.getString("kafka.broker.list"))
    props.put("acks","1")
    props.put("retries","1")
    props.put("buffer.memory"," 33554432")
    props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer")
    props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer")

    val producer: Producer[String,String] = new KafkaProducer(props)

    producer.send(new ProducerRecord[String,String]("test",0,"axx","14:39:43.358 [main] INFO  com.flyex.AppMain - 1607323183358|{\"cm\":{\"ln\":\"-113.4\",\"sv\":\"V2.3.0\",\"os\":\"8.1.7\",\"g\":\"7BZE66H5@gmail.com\",\"mid\":\"m305\",\"nw\":\"WIFI\",\"l\":\"pt\",\"vc\":\"0\",\"hw\":\"1080*1920\",\"ar\":\"MX\",\"uid\":\"u643\",\"t\":\"1607240453416\",\"la\":\"11.4\",\"md\":\"sumsung-19\",\"vn\":\"1.3.4\",\"ba\":\"Sumsung\",\"sr\":\"Z\"},\"ap\":\"com.flyex.AppMain\",\"et\":[{\"ett\":\"1607273557535\",\"en\":\"start\",\"kv\":{\"entry\":\"2\",\"loading_time\":\"12\",\"action\":\"1\",\"open_ad_type\":\"2\",\"detail\":\"\"}}]}        2020-12-07"))

    producer.close()
  }

}
